docs/proposals/improve-status-in-CAPI-resources.md

```diff
@@ -0,0 +1,921 @@
+    // +listMapKey=type
+    Conditions []metav1.Condition `json:"conditions,omitempty"`
+
+    // Other fields...
+    // NOTE: `FailureReason`, `FailureMessage` fields won't be there anymore
+}
+```
+
+| v1beta1 (current)                | v1beta2 (tentative Q1 2025)                              | v1beta1 removal (tentative Q1 2026) |
+|----------------------------------|----------------------------------------------------------|-------------------------------------|
+|                                  | `BackCompatibilty` (new)                                 | (removed)                           |
+| `ReadyReplicas` (deprecated)     | `BackCompatibilty.ReadyReplicas` (renamed) (deprecated)  | (removed)                           |
+| `ExprimentalReadyReplicas` (new) | `ReadyReplicas` (renamed)                                | `ReadyReplicas`                     |
+| `FailureReason` (deprecated)     | `BackCompatibilty.FailureReason` (renamed) (deprecated)  | (removed)                           |
+| `FailureMessage` (deprecated)    | `BackCompatibilty.FailureMessage` (renamed) (deprecated) | (removed)                           |
+| `Conditions`                     | `BackCompatibilty.Conditions` (renamed) (deprecated)     | (removed)                           |
+| `ExprimentalConditions` (new)    | `Conditions` (renamed)                                   | `Conditions`                        |
+| `UpToDateReplicas` (new)         | `UpToDateReplicas`                                       | `UpToDateReplicas`                  |
+| other fields...                  | other fields...                                          | other fields...                     |
+
+Notes:
+- The `BackCompatibilty` struct is going to exist in v1beta2 types only until v1beta1 removal (9 months or 3 minor releases after v1beta2 is released/v1beta1 is deprecated, whichever is longer).
+  Fields in this struct are used for supporting down conversions, thus providing users relying on v1beta1 APIs additional buffer time to pick up the new changes.
+- This proposal is using `UpToDateReplicas` instead of `UpdatedReplicas`; This is a deliberated choice to avoid 
+  confusion between update (any change) and upgrade (change of the Kubernetes versions).
+- Also `AvailableReplicas` will determine Machine's availability by reading Machine.Available condition instead of
+  computing availability as of today, however in this case the semantic of the field is not changed
+
+TODO: check `FullyLabeledReplicas`, do we still need it?
+
+#### MachineSet (New)Conditions
+
+| Condition        | Note                                                                                                           |
+|------------------|----------------------------------------------------------------------------------------------------------------|
+| `ReplicaFailure` | This condition surfaces issues on creating Machines, if any.                                                   |
```

Comment:
It's not obvious to me the diff between ReplicaFailure and MachinesReady

Reply:
Improved the description, PTAL
FYI The idea comes from NewReplicaSetAvailable in Kubernetes, but I don't feel strongly about this condition and we might also drop it entirely